src/pkg/leaderElection.go

```diff
@@ -43,8 +44,26 @@ func RunLeaderElection(client *clientset.Clientset, lockName, lockIdentity, lock
 				log.Info().Msgf("Perform Migration")
 				for {
 					log.Info().Msgf("leader is %s", lockIdentity)
-					log.Info().Msgf("Getting replica count...")
-					time.Sleep(5 * time.Second)
+					replicaCount, err := getReplicas()
+					if err != nil {
+						log.Fatal().Msgf("Failed to get replica count")
+					}
+					log.Info().Msgf("Setting replica count to %v", replicaCount)
+					retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+						deploymentsClient := client.AppsV1().Deployments(lockNamespace)
+						result, getErr := deploymentsClient.Get(context.TODO(), lockName, metav1.GetOptions{})
+						if getErr != nil {
+							log.Fatal().Msgf("Failed to get latest version of Deployment: %v", getErr)
+						}
+						result.Spec.Replicas = &replicaCount
+						_, updateErr := deploymentsClient.Update(context.TODO(), result, metav1.UpdateOptions{})
+						return updateErr
+					})
+					if retryErr != nil {
+						log.Fatal().Msgf("Failed to set replica count: %v", retryErr)
+					}
+					log.Info().Msgf("Successfully set replicas to %v", replicaCount)
+					time.Sleep(60 * time.Second)
```

Comment:
should this be configurable?

Reply:
Yeah, I was planning on adding a flag for this but wasn't sure if it we wanted that since that might allow self-hosted customers to set it too low and bog us down with API calls?

Reply:
Good point - ok lets leave that for now and put a code comment here about that.